searches/linear_search.nim

```diff
@@ -0,0 +1,88 @@
+## Linear Search
+## =============
+## Linear search is the simplest but least efficient searching algorithm
+## to search for an element in a data set.
```

Comment:
data set being an array here. The algorithm could apply to a linked list too, or anything that could be sequentially indexed, well anything that can be stored in an array.
```suggestion
## to search for an element in an array.
```